plugins/modules/ipahost.py

```diff
@@ -1337,6 +1342,8 @@ def main():
             else:
                 ansible_module.fail_json(msg="Unkown state '%s'" % state)
 
+        del host_set
```

Comment:
Why deleting the set? This will only decrement ref counter and will not immediately free memory (even if it did it wouldn't be a considerable amount of memory).

Reply:
This could be a set with thousands of entries.

Reply:
The set will contain all names, also the ones that are not touched in the second stage. Also the information that is passed to module might already be a lot (think about 10000+ users or hosts with lots of settings). Do you prefer a host_set.clear() instead?

Reply:
I don't get the problem.
With the del statement we loose nothing, but let Python know we don't need the objects anymore, and we are about to run commands in the IPA API, which will consume memory. If we have few objects the impact is negligible, if we have tens of thousands objects, we will allow the removal of objects if needed.
Why is it not good? (Really, I don't know and want to learn something today.)